mlflow/store/job/abstract_store.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABCMeta, abstractmethod
+from typing import Iterator
+
+from mlflow.entities._job import Job
+from mlflow.entities._job_status import JobStatus
+from mlflow.utils.annotations import developer_stable
+
+
+@developer_stable
+class AbstractJobStore:
+    """
+    Abstract class that defines API interfaces for storing Job metadata.
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        """
+        Empty constructor. This is deliberately not marked as abstract, else every derived class
+        would be forced to create one.
```

Comment:
looks like nothing enforces this

Reply:
it is fine to copy and paste code but there is no guarantee that the original code is correct.